PaperBlossoms/data/README.md

```diff
@@ -0,0 +1,312 @@
+
+### Scripts
+
+A handful of [helper scripts](scripts/) automate common tasks. These tasks fall into two categories: tasks which are part of the data entry workflow and tasks which update the data entry system. 
+
+The general data entry workflow is as follows:
+
+1. Modify JSON file(s): manual
+2. Validate JSON file(s): [validate_json.py](scripts/validate_json.py)
+3. Export data from JSON files to `paperblossoms.db`: [json_to_db.py](scripts/json_to_db.py)
+
+`validate_json.py` loops through schemas found in the `json_schema/` directory and validates the JSON of the same name in the `json/` directory. It will raise an informative error if a JSON fails validation. This script takes no arguments, and outputs the schema name as it loops.
+
+```
+$ python3 PaperBlossoms/data/scripts/validate_json.py 
+Validated qualities.json
+Validated armor.json
+Validated techniques.json
+Validated bonds.json
+Validated personal_effects.json
+Validated titles.json
+Validated samurai_heritage.json
+Validated schools.json
+Validated item_patterns.json
+Validated upbringings.json
+Validated rings.json
+Validated skill_groups.json
+Validated regions.json
+Validated clans.json
+Validated weapons.json
+Validated advantages_disadvantages.json
+```
+
+`json_to_db.py` is the workhorse script for converting JSON data into SQLite data. It constructs the entire SQLite database, including base data, custom data, and translations. It loads data from each JSON and writes that data to SQLite tables in a flattened and normalized way. It takes no arguments and doesn't write to stdout.
+
+```
+$ python3 PaperBlossoms/data/scripts/json_to_db.py
+```
+
+Other scripts modify the data entry system via the JSON schemas.
+
+`add_default_snippets_to_schema.py` adds or overwrites default snippets to a specified JSON schema based on the schema. This script is intended to be used after a new schema is written or an existing schema is updated. Rather than manually adding or updating the default snippets, use this script to automatically generate those snippets. They can be modified afterwards in order to set default values, remove unwanted fields, etc. This script takes one argument, the JSON schema file, and will add default snippets for all objects in the JSON schema, no matter the level of nesting.
+
+```
+$ python3 PaperBlossoms/data/scripts/add_default_snippets_to_schema.py PaperBlossoms/data/json_schema/bonds.schema.json
+```
+
+`add_enums.py` adds or overwrites `enum` lists in JSON schemas based on entered data (primarily JSON). For example, to generate a valid list of skills and techniques for school curriculum advancements, this script will load `skills.json` and `techniques.json`, extract the names of all skills and techniques, and write those names as the `enum` list for `advance` in `curriculum` within `schools.json`. This script is intended to be used when an `enum` list needs to be refreshed (new techniques, new books, etc). It takes the following arguments:
```

Comment:
This feels like a good place to reiterate (again, possibly in more detail) what enums are and why enums should be updated: not because they are needed by the application, but because by keeping them updated, consistent typo-free data entry will be much more feasible.